src/network/core/address.cpp

```diff
@@ -239,7 +239,18 @@ SOCKET NetworkAddress::Resolve(int family, int socktype, int flags, SocketList *
 		strecpy(this->hostname, fam == AF_INET ? "0.0.0.0" : "::", lastof(this->hostname));
 	}
 
+	static bool _resolve_timeout_error_message_shown = false;
+	auto start = std::chrono::steady_clock::now();
 	int e = getaddrinfo(StrEmpty(this->hostname) ? nullptr : this->hostname, port_name, &hints, &ai);
+	auto end = std::chrono::steady_clock::now();
+	std::chrono::duration<double> duration = end - start;
+	if (!_resolve_timeout_error_message_shown && duration.count() > 5) {
```

Comment:
what I personally found an easier way of reading this, is something like this:

`end - start > std::chrono::seconds(5)`

That makes it more clear what the time unit is, and allows easier changing that :)

Reply:
The problem is that I need the number of seconds later again. I have messed with the code itself so it is clearer that they are actually seconds.